### ISSUE TITLE
Issues using different language servers for different C++ projects at the same time

### ISSUE THREAD

**Issue by opti-mix** [neoclide/coc.nvim#2329]
I'd like to be able to work with multiple C++ projects in the same nvim instance at the same time and switch between files from those projects at any time. Additionally, I configured different project local configs in .vim/coc-settings.json files as for various reasons I need to use different language servers configs for different C++ projects. 

I have two projects A and B. If I first open a C++ file from A, I can see that a proper language server is started and all the functionality is there. Then I open a file from B. And I can see that a proper language  server for B is started. But I also see that the language server for A tries to process this file from B as well. As a result, I get a bunch of errors/diagnostics markers all over the place produced by the language server for A when it processes a file from B. When I try to do `CocList symbols` on one of the C++ files, it seems like coc tries to pull symbol information from both projects at the same time, i.e. from both language servers.

So, overall, it seems like instead of using only a single language server related to a given buffer/file belonging to a project, coc tries to pull information from ALL language servers, probably based on the fact that they can  handle files of a given type, e.g. C++ files, and does not respect the fact that a specific instance of a language server was configured for each of the projects.

I'm trying to understand if it is possible to have a multi-project setup using coc with a clear separation of language servers and all the functionality based on them between projects, like I described in the beginning. I'm not sure if I haven't configured coc properly or if such a setup is basically not supported by coc yet.

Any help is appreciated! Thanks!

**Comment by chemzqm** [comment]
Use workspace folder configuration file https://github.com/neoclide/coc.nvim/wiki/Using-the-configuration-file#configuration-file-resolve

**Comment by opti-mix** [comment]
@chemzqm 

> Use workspace folder configuration file https://github.com/neoclide/coc.nvim/wiki/Using-the-configuration-file#configuration-file-resolve

Sorry, but as I indicated in my explanation, I **do use** workspace configuration files already and they are saved in `.vim/coc-settings.json` under the root of each workplace! This is how both of my projects A and B are recognized as workplaces  and coc loads different configurations for them and starts different language servers for them. I checked this by doing: `echo g:WorkspaceFolders` and after I open files from both projects, it shows two correct workspace roots.

My problem is that when I use files from both workspaces in the same nvim editing session, it seems like **coc uses 2 languages servers at the same time** for each file being edited or for coc lists like `symbols`. I expected coc to use to use only one of the language servers for each file, configured for the corresponding workspaces this file belongs to.
